File: Easy/lumen/lumen.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;


class Solution
{
    static void Main(string[] args)
    {
        int N = int.Parse(Console.ReadLine());
        int L = int.Parse(Console.ReadLine());
        List<List<int>> cells = new List<List<int>>();
        List<List<int>> candles = new List<List<int>>();
        for (int i = 0; i < N; i++)
        {
            string[] inputs = Console.ReadLine().Split(' ');
            List<int> rowHold = new List<int>();
            for (int j = 0; j < N; j++)
            {
                if (inputs[j] == "C")
                {
                    candles.Add(new List<int>{j, i});
                }
                rowHold.Add(0);
            }
            cells.Add(rowHold);
        }

        candles.ForEach(candle => {
            for (int i = 0; i < N; i++)
            {
                for (int j = 0; j < N; j++)
                {
                    cells[j][i] = Math.Max(cells[j][i], Math.Max(0, L - Math.Max(Math.Abs(candle[0]-j), Math.Abs(candle[1]-i))));
                }
            }
        });

        int darkCount = 0;

        cells.ForEach(row => row.ForEach(cell => {
            if (cell == 0)
            {
                darkCount++;
            }
        }));

        Console.WriteLine(darkCount);
    }
} 

File: Easy/lumen/lumen.js
const N = parseInt(readline());
const L = parseInt(readline());
let cells = [];
let candles = [];

for (let i = 0; i < N; i++) {
    var inputs = readline().split(' ');
    let rowHold = [];
    for (let j = 0; j < N; j++) {
        if (inputs[j] === "C") candles.push([j, i]);
        rowHold.push(0);
    }
    cells.push(rowHold);
}

candles.forEach(candle => {
    for (let i = 0; i < N; i++){
        for (let j = 0; j < N; j++){
            cells[j][i] = Math.max(cells[j][i], Math.max(0, L - Math.max(Math.abs(candle[0]-j), Math.abs(candle[1]-i))));

        }
    }
});

let darkCount = 0;

cells.forEach(row => row.forEach(cell => {
    if (cell === 0) darkCount++;
}));

console.log(darkCount); 
